flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/expressions/converter/CustomizedConvertRule.java

```diff
@@ -0,0 +1,361 @@
+		DEFINITION_RULE_MAP.put(BuiltInFunctionDefinitions.PLUS, CustomizedConvertRule::convertPlus);
+		DEFINITION_RULE_MAP.put(BuiltInFunctionDefinitions.CEIL, CustomizedConvertRule::convertCeil);
+		DEFINITION_RULE_MAP.put(BuiltInFunctionDefinitions.FLOOR, CustomizedConvertRule::convertFloor);
+		DEFINITION_RULE_MAP.put(BuiltInFunctionDefinitions.TEMPORAL_OVERLAPS, CustomizedConvertRule::convertTemporalOverlaps);
+		DEFINITION_RULE_MAP.put(BuiltInFunctionDefinitions.TIMESTAMP_DIFF, CustomizedConvertRule::convertTimestampDiff);
+		DEFINITION_RULE_MAP.put(BuiltInFunctionDefinitions.ARRAY, CustomizedConvertRule::convertArray);
+		DEFINITION_RULE_MAP.put(BuiltInFunctionDefinitions.ARRAY_ELEMENT, CustomizedConvertRule::convertArrayElement);
+		DEFINITION_RULE_MAP.put(BuiltInFunctionDefinitions.MAP, CustomizedConvertRule::convertMap);
+		DEFINITION_RULE_MAP.put(BuiltInFunctionDefinitions.ROW, CustomizedConvertRule::convertRow);
+		DEFINITION_RULE_MAP.put(BuiltInFunctionDefinitions.ORDER_ASC, CustomizedConvertRule::convertOrderAsc);
+
+		DEFINITION_RULE_MAP.put(BuiltInFunctionDefinitions.SQRT, (children, context) ->
+			context.getRelBuilder().call(FlinkSqlOperatorTable.POWER, context.toRexNodes(
+				Arrays.asList(children.get(0), ApiExpressionUtils.valueLiteral(0.5)))));
+
+		// blink expression
+		DEFINITION_RULE_MAP.put(InternalFunctionDefinitions.THROW_EXCEPTION, (children, context) -> {
+			DataType type = ((TypeLiteralExpression) children.get(1)).getOutputDataType();
+			SqlThrowExceptionFunction function = new SqlThrowExceptionFunction(
+					context.getTypeFactory().createFieldTypeFromLogicalType(fromDataTypeToLogicalType(type)));
+			return context.getRelBuilder().call(function, context.toRexNode(children.get(0)));
+		});
+	}
+
+	@Override
+	public Optional<RexNode> convert(CallExpression call, ConvertContext context) {
+		Conversion conversion = DEFINITION_RULE_MAP.get(call.getFunctionDefinition());
+		if (conversion == null) {
+			return Optional.empty();
+		} else {
+			return Optional.of(conversion.convert(call.getChildren(), context));
+		}
+	}
+
+	private static RexNode convertCast(List<Expression> children, ConvertContext context) {
+		RexNode child = context.toRexNode(children.get(0));
+		TypeLiteralExpression type = (TypeLiteralExpression) children.get(1);
+		return context.getRelBuilder().getRexBuilder().makeAbstractCast(
+			context.getTypeFactory().createFieldTypeFromLogicalType(
+				type.getOutputDataType().getLogicalType().copy(child.getType().isNullable())),
+			child);
+	}
+
+	private static RexNode convertArrayElement(List<Expression> children, ConvertContext context) {
+		List<RexNode> childrenRexNode = context.toRexNodes(children);
+		return context.getRelBuilder().call(FlinkSqlOperatorTable.ELEMENT, childrenRexNode);
+	}
+
+	private static RexNode convertOrderAsc(List<Expression> children, ConvertContext context) {
+		List<RexNode> childrenRexNode = context.toRexNodes(children);
+		return childrenRexNode.get(0);
```

Comment:
Can we assert for the number of arguments? Similar as in `PlannerExpressionConverter`? It actually applies to all the Conversions.

Reply:
The idea was to let calcite assert, but active assert would be better.